src/test/java/uk/gov/di/ipv/cri/common/library/domain/personidentity/AddressTest.java

```diff
@@ -9,24 +9,74 @@
 
 class AddressTest {
     @Test
-    void shouldGetPastAddressType() {
+    void shouldGetPreviousAddressTypeWithJustValidUntil() {
         Address testAddress = new Address();
         testAddress.setValidUntil(LocalDate.of(2013, 8, 9));
 
         assertEquals(AddressType.PREVIOUS, testAddress.getAddressType());
     }
 
     @Test
-    void shouldGetCurrentAddressType() {
+    void shouldGetPreviousAddressTypeWhenUntilIsToday() {
+        Address testAddress = new Address();
+        testAddress.setValidUntil(LocalDate.now().atTime(0, 0).toLocalDate());
+
+        assertEquals(AddressType.PREVIOUS, testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldGetPreviousAddressTypeWithValidUntilAndValidFrom() {
+        Address testAddress = new Address();
+        testAddress.setValidFrom(LocalDate.of(2013, 7, 9));
+        testAddress.setValidUntil(LocalDate.of(2013, 8, 9));
+
+        assertEquals(AddressType.PREVIOUS, testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldGetCurrentAddressTypeWithJustValidFrom() {
         Address testAddress = new Address();
         testAddress.setValidFrom(LocalDate.of(2021, 9, 10));
 
         assertEquals(AddressType.CURRENT, testAddress.getAddressType());
     }
 
     @Test
-    void shouldReturnNullAddressTypeWhenValidFromAndValidUntilNotPresent() {
+    void shouldReturnCurrentAddressTypeWhenValidFromAndValidUntilAreNull() {
+        Address testAddress = new Address();
+
+        assertEquals(AddressType.CURRENT, testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldReturnCurrentAddressTypeWhenValidFromIsToday() {
+        Address testAddress = new Address();
+        testAddress.setValidFrom(LocalDate.now().atTime(0, 0).toLocalDate());
```

Comment:
The `LocalDate` instances that will be received by the CRIs won't have time specified.
Therefore, it should not be necessary to include `atTime()` in any of the unit tests